carbon-design-system/carbon
# feat(TimePicker): refactor to functional component

REF #9712 

Refactored `TimePicker` and `TimePickerSelect` into functional components.

#### Changelog

**New**

- `TimePickerNext` and `TimePickerSelectNext`
- Added RTL for `TimePickerNext`.

**Changed** (in both Classic and Next versions)

- Removed `labelText` and `iconDescription` knobs from `TimePicker` story.
- Removed `hideLabel` prop and `label` logic from `TimePickerSelect`.
- Removed deprecated iconDescription from `TimePickerSelect`. 
- Added aria-label to the select element in `TimePickerSelect`.
- Added aria-hidden to the icon `TimePickerSelect`.

**Questions**

- `TimePicker` has some `evt.persist();` in it. Unclear if that needs to be removed based on comments in [9535](https://github.com/carbon-design-system/carbon/issues/9535)

- Can’t get the new icons in `TimePickerSelect` to import without breaking the Netlify build.

#### Testing / Reviewing

1. Pull down locally
2. Go to `packages/react`
3. Run `yarn start:v11` to run old storybook with v11 components.
4. Ensure tests for Classic and Next versions of `TimePicker` go green.
